include/net/net_pkt.h

```diff
@@ -98,6 +98,10 @@ struct net_pkt {
 #if defined(CONFIG_NET_PKT_TIMESTAMP)
 	/** Timestamp if available. */
 	struct net_ptp_time timestamp;
+#endif
+#if defined(CONFIG_NET_PKT_TXTIME)
+	/** Network packet TX time in the future (in nanoseconds) */
+	u64_t txtime;
```

Comment:
This can probably combined with the timestamp field so we can save some space, I will rework this part.